File: Templates/Java/src/test/java/KataTest.java
import static org.junit.jupiter.api.Assertions.assertEquals;

import java.io.IOException;
import java.util.Arrays;
import java.util.List;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

public class KataTest {
    private Kata fixture;

    @BeforeEach
    public void setUp() {
        fixture = new Kata();
    }

    @Test
    public void helloShouldReturnMessageWithSpecifiedName() {
        String name = "Ingage";
        assertEquals(String.format("Hello %s", name), fixture.hello(name));
    }
    
    @Test
    public void smallMapShortestPathShouldBeCorrect() throws IOException {
        char[][] smallMap = fixture.loadMap("../../cornmaze-small.txt");
        List<Point> smallPath = fixture.findShortestPath(smallMap);
        
        List<Point> expected = Arrays.asList(
            new Point(1, 0), new Point(1, 1), new Point(1, 2), new Point(1, 3), 
            new Point(1, 4), new Point(1, 5), new Point(1, 6), new Point(1, 7), 
            new Point(2, 7), new Point(3, 7), new Point(3, 8), new Point(4, 8), 
            new Point(5, 8), new Point(6, 8), new Point(7, 8), new Point(8, 8), 
            new Point(8, 9)
        );
        
        assertEquals(expected, smallPath);
    }
    
    @Test
    public void largeMapShortestPathShouldBeCorrect() throws IOException {
        char[][] largeMap = fixture.loadMap("../../cornmaze-large.txt");
        List<Point> largePath = fixture.findShortestPath(largeMap);
        
        List<Point> expected = Arrays.asList(
            new Point(13, 0), new Point(13, 1), new Point(14, 1), new Point(15, 1), 
            new Point(15, 2), new Point(15, 3), new Point(16, 3), new Point(17, 3),
            new Point(17, 2), new Point(17, 1), new Point(18, 1), new Point(19, 1), 
            new Point(20, 1), new Point(21, 1), new Point(22, 1), new Point(23, 1),
            new Point(24, 1), new Point(25, 1), new Point(25, 2), new Point(25, 3), 
            new Point(25, 4), new Point(25, 5), new Point(25, 6), new Point(25, 7),
            new Point(25, 8), new Point(25, 9), new Point(25, 10), new Point(25, 11), 
            new Point(25, 12), new Point(25, 13), new Point(26, 13), new Point(27, 13),
            new Point(28, 13), new Point(29, 13), new Point(29, 12), new Point(29, 11), 
            new Point(28, 11), new Point(27, 11), new Point(27, 10), new Point(27, 9),
            new Point(27, 8), new Point(27, 7), new Point(27, 6), new Point(27, 5), 
            new Point(27, 4), new Point(27, 3), new Point(27, 2), new Point(27, 1),
            new Point(28, 1), new Point(29, 1), new Point(30, 1), new Point(31, 1), 
            new Point(31, 2), new Point(31, 3), new Point(30, 3), new Point(29, 3),
            new Point(29, 4), new Point(29, 5), new Point(29, 6), new Point(29, 7), 
            new Point(29, 8), new Point(29, 9), new Point(30, 9), new Point(31, 9),
            new Point(31, 10), new Point(31, 11), new Point(31, 12), new Point(31, 13), 
            new Point(31, 14), new Point(31, 15), new Point(31, 16), new Point(31, 17),
            new Point(31, 18), new Point(31, 19), new Point(31, 20), new Point(31, 21), 
            new Point(31, 22), new Point(31, 23), new Point(30, 23), new Point(29, 23),
            new Point(29, 24), new Point(29, 25), new Point(29, 26), new Point(29, 27), 
            new Point(29, 28), new Point(29, 29), new Point(30, 29), new Point(31, 29),
            new Point(31, 30), new Point(31, 31), new Point(31, 32)
        );
        
        assertEquals(expected, largePath);
    }
}

File: Templates/Kotlin/src/test/kotlin/KataTest.kt
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class KataTest {
    private lateinit var fixture: Kata

    @BeforeEach
    fun setUp() {
        fixture = Kata()
    }

    @Test
    fun `hello should return message with specified name`() {
        val name = "Ingage"
        assertEquals("Hello $name", fixture.hello(name))
    }
    
    @Test
    fun `small map shortest path should be correct`() {
        val smallMap = fixture.loadMap("../../cornmaze-small.txt")
        val smallPath = fixture.findShortestPath(smallMap)
        
        val expected = listOf(
            Point(1, 0), Point(1, 1), Point(1, 2), Point(1, 3),
            Point(1, 4), Point(1, 5), Point(1, 6), Point(1, 7),
            Point(2, 7), Point(3, 7), Point(3, 8), Point(4, 8),
            Point(5, 8), Point(6, 8), Point(7, 8), Point(8, 8),
            Point(8, 9)
        )
        
        assertEquals(expected, smallPath)
    }
    
    @Test
    fun `large map shortest path should be correct`() {
        val largeMap = fixture.loadMap("../../cornmaze-large.txt")
        val largePath = fixture.findShortestPath(largeMap)
        
        val expected = listOf(
            Point(13, 0), Point(13, 1), Point(14, 1), Point(15, 1),
            Point(15, 2), Point(15, 3), Point(16, 3), Point(17, 3),
            Point(17, 2), Point(17, 1), Point(18, 1), Point(19, 1),
            Point(20, 1), Point(21, 1), Point(22, 1), Point(23, 1),
            Point(24, 1), Point(25, 1), Point(25, 2), Point(25, 3),
            Point(25, 4), Point(25, 5), Point(25, 6), Point(25, 7),
            Point(25, 8), Point(25, 9), Point(25, 10), Point(25, 11),
            Point(25, 12), Point(25, 13), Point(26, 13), Point(27, 13),
            Point(28, 13), Point(29, 13), Point(29, 12), Point(29, 11),
            Point(28, 11), Point(27, 11), Point(27, 10), Point(27, 9),
            Point(27, 8), Point(27, 7), Point(27, 6), Point(27, 5),
            Point(27, 4), Point(27, 3), Point(27, 2), Point(27, 1),
            Point(28, 1), Point(29, 1), Point(30, 1), Point(31, 1),
            Point(31, 2), Point(31, 3), Point(30, 3), Point(29, 3),
            Point(29, 4), Point(29, 5), Point(29, 6), Point(29, 7),
            Point(29, 8), Point(29, 9), Point(30, 9), Point(31, 9),
            Point(31, 10), Point(31, 11), Point(31, 12), Point(31, 13),
            Point(31, 14), Point(31, 15), Point(31, 16), Point(31, 17),
            Point(31, 18), Point(31, 19), Point(31, 20), Point(31, 21),
            Point(31, 22), Point(31, 23), Point(30, 23), Point(29, 23),
            Point(29, 24), Point(29, 25), Point(29, 26), Point(29, 27),
            Point(29, 28), Point(29, 29), Point(30, 29), Point(31, 29),
            Point(31, 30), Point(31, 31), Point(31, 32)
        )
        
        assertEquals(expected, largePath)
    }
}
